src/dbxdeploy/package/PackageDeployer.py

```diff
@@ -3,50 +3,101 @@
         logger: Logger,
         packageUploader: PackageUploaderInterface,
         packageBuilder: PackageBuilder,
         targetPathsResolver: TargetPathsResolver,
+        localPathsResolver: LocalPathsResolver,
     ):
         self.__projectBaseDir = projectBaseDir
+        self.__offlineInstall = offlineInstall
         self.__logger = logger
         self.__packageUploader = packageUploader
         self.__packageBuilder = packageBuilder
         self.__targetPathsResolver = targetPathsResolver
+        self.__localPathsResolver = localPathsResolver
 
     def deploy(self, packageMetadata: PackageMetadata):
-        whlPath = self.__targetPathsResolver.getPackageUploadPathForDeploy(packageMetadata)
-
         def whlContentReadyCallback(content):
-            self.__upload(content, whlPath)
+            if self.__offlineInstall:
+                self.__checkOfflineDependencies(packageMetadata)
+
+                for dependency in packageMetadata.dependencies:
+                    dependencyLocalPath = self.__localPathsResolver.getDependencyLocalPath(dependency)
+                    dependencyFilename = self.__localPathsResolver.getDependencyFilename(dependency)
+                    dependencyDeployPath = self.__targetPathsResolver.getDependencyUploadPathForDeploy(packageMetadata, dependencyFilename)
+
+                    with dependencyLocalPath.open('rb') as file:
+                        _content = file.read()
+                        self.__upload(_content, dependencyDeployPath)
+
+                requitementsLocalPath = self.__localPathsResolver.getRequirementsLocalPath()
+                requitementsDeployPath = self.__targetPathsResolver.getRequirementsUploadPathForDeploy(packageMetadata)
+
+                with requitementsLocalPath.open('rb') as file:
+                    _content = file.read()
+                    self.__upload(_content, requitementsDeployPath)
+
+            masterPackageDeployPath = self.__targetPathsResolver.getPackageUploadPathForDeploy(packageMetadata)
+            self.__upload(content, masterPackageDeployPath)
 
         self.__invoke(packageMetadata, whlContentReadyCallback)
 
     def release(self, packageMetadata: PackageMetadata):
-        whlPath = self.__targetPathsResolver.getPackageUploadPathForRelease(packageMetadata)
-
         def whlContentReadyCallback(content):
-            self.__upload(content, whlPath)
+            if self.__offlineInstall:
+                self.__checkOfflineDependencies(packageMetadata)
+
+                for dependency in packageMetadata.dependencies:
+                    dependencyLocalPath = self.__localPathsResolver.getDependencyLocalPath(dependency)
+                    dependencyFilename = self.__localPathsResolver.getDependencyFilename(dependency)
+                    dependencyReleasePath = self.__targetPathsResolver.getDependencyUploadPathForRelease(packageMetadata, dependencyFilename)
+
+                    with dependencyLocalPath.open('rb') as file:
```

Comment:
Duplicate block (similar code bellow)

Reply:
Aware of that. Was a quick and dirty solution to make it work. Probably it would be best to create class PackageDependencyDeployer and move logic there.

Reply:
check https://github.com/bricksflow/dbx-deploy/pull/4/commits/5a309385611e4970554bcb46cc1e27c475afd57c